File: TestRoundingKotlin.java
public class TestRoundingKotlin {
    public static void main(String[] args) {
        // Red Blob Games article explicitly mentions Kotlin's Math.round
        // "Math.round() in Kotlin uses round-to-even" -> no, kotlin.math.round does.
        // java.lang.Math.round rounds half-up.
        // Let's replace the script to test round-to-even.
        System.out.println("Nothing broken about Math.round algorithm itself.");
    }
}

File: TestRoundingKotlin.kt
import kotlin.math.*

fun main() {
    val kotlinRound05 = kotlin.math.round(0.5)
    val kotlinRoundMinus05 = kotlin.math.round(-0.5)

    val javaRound05 = java.lang.Math.round(0.5)
    val javaRoundMinus05 = java.lang.Math.round(-0.5)

    println("Kotlin round(0.5): \$kotlinRound05") // expected: 0.0
    println("Kotlin round(-0.5): \$kotlinRoundMinus05") // expected: -0.0

    println("Java round(0.5): \$javaRound05") // expected: 1
    println("Java round(-0.5): \$javaRoundMinus05") // expected: 0
}
